tests/utils.ts

```diff
@@ -0,0 +1,377 @@
+import {
+  fromFileUrl,
+  parse,
+  relative,
+  resolve,
+} from "https://deno.land/std@0.182.0/path/mod.ts";
+import {
+  ensureFileSync,
+  existsSync,
+} from "https://deno.land/std@0.182.0/fs/mod.ts";
+import { AssertionError } from "https://deno.land/std@0.182.0/testing/asserts.ts";
+import {
+  buildMessage,
+  diffstr,
+} from "https://deno.land/std@0.182.0/testing/_diff.ts";
+import { bundle, type BundleOptions, emit, type EmitOptions } from "../mod.ts";
+import * as base64Url from "https://deno.land/std@0.182.0/encoding/base64url.ts";
+import * as base64 from "https://deno.land/std@0.182.0/encoding/base64.ts";
+
+const textEncoder = new TextEncoder();
+const textDecoder = new TextDecoder();
+
+const inlineSourceMapRegex =
+  /^\/\/# sourceMappingURL=data:application\/json;base64,([a-zA-Z0-9+/=]+)$/;
+
+// Ensures that operations that update the snapshots are performed sequentially,
+// in order, and at the end of the test run.
+const updateQueue = createUpdateQueue();
+
+type TranspileResult = Awaited<ReturnType<typeof emit>>;
+type BundleResult = Awaited<ReturnType<typeof bundle>>;
+
+const osType = typeof Deno?.build?.os === "string" ? Deno.build.os : "linux";
+
+const EOL = osType === "windows" ? "\r\n" : "\n";
```

Comment:
I have added this bit and used it everywhere instead of `\n`. @dsherret Does that make the tests pass now?

Reply:
I had fixed that issue by adding a `.gitattributes` to this PR, which is preferable because then we don't need these kind of conditions. The tests were complaining about this:

```
,-     "file:///D:/a/deno_emit/deno_emit/testdata/mod.ts": "cYfK3r_ND0Q_nXVQYOxJjg8VYMM.js"\n
,+     "file:///testdata/mod.ts": "0BJ1Cp_wwHSv6YnJzSlY2UlxpSc.js"\n
```

Previous run: https://github.com/denoland/deno_emit/actions/runs/4726314879/jobs/8385766439

Now the tests are failing with more newline stuff. Can you revert your last change (I have it locally in case you've lost it so just let me know if you want me to push it to this PR)? Also, please land commits on top of each other instead of overwritting commits because I don't know what has changed since the last time I reviewed. We'll squash all the commits when merging.

Reply:
I reverted the changes. I also found how to run the Actions on my own fork, so I can iterate on the Windows issue.

(FYI, the Windows build takes a _lot_ of time. Compiling Rust dependencies takes more than 12 minutes, whereas it takes less than 5 minutes for Linux and less than 2 minutes for macOS.)